v1/protocol_binding/pubsub_binder.cc

```diff
@@ -0,0 +1,120 @@
+#include "pubsub_binder.h"
+
+namespace cloudevents {
+namespace binding {
+
+using ::google::pubsub::v1::PubsubMessage;
+using ::io::cloudevents::v1::CloudEvent;
+using ::io::cloudevents::v1::CloudEvent_CloudEventAttribute;
+using ::cloudevents::format::StructuredCloudEvent;
+using ::cloudevents::format::Format;
+using ::cloudevents::format::Formatter;
+using ::cloudevents::formatter_util::FormatterUtil;
+using ::cloudevents::cloudevents_util::CloudEventsUtil;
+
+typedef absl::flat_hash_map<std::string, CloudEvent_CloudEventAttribute>
+  CeAttrMap;
+
+inline static constexpr char kPubsubContentKey[] = "content-type";
+
+
+template <>
+absl::StatusOr<std::string> Binder<PubsubMessage>::GetContentType(
+    const PubsubMessage& pubsub_msg) {
+  google::protobuf::Map<std::string, std::string> attrs =
+    pubsub_msg.attributes();
+  auto ind = attrs.find(kPubsubContentKey);
+  if (ind == attrs.end()) {
+    return std::string("");
+  }
+  return ind -> second;
+}
+
+template <>
+absl::StatusOr<std::string> Binder<PubsubMessage>::GetPayload(
+    const PubsubMessage& pubsub_msg) {
+  // get payload and base64 decode
+  return base64::base64_decode(pubsub_msg.data());
+}
+
+template <>
+absl::StatusOr<CloudEvent> Binder<PubsubMessage>::UnbindBinary(
+    const PubsubMessage& pubsub_msg) {
+  CloudEvent cloud_event;
+
+  for (auto const& attr : pubsub_msg.attributes()) {
+    std::string key;
+    if (attr.first == kPubsubContentKey) {
+      key = kContenttypeKey;
+    } else if (attr.first.rfind(kMetadataPrefix, 0) == 0){
+      size_t len_prefix = strlen(kMetadataPrefix);
```

Comment:
strlen is a C-style function that computes length in linear time, is there any way to make kMetadataPrefix string or string_view to have compile time length or O(1) length.